kale/utils/metadata_utils.py

```diff
@@ -0,0 +1,103 @@
+import re
+
+from kale.utils.pod_utils import is_workspace_dir
+
+METADATA_REQUIRED_KEYS = [
+    'experiment_name',
+    'pipeline_name',
+]
+kale_step_name_regex = r'^[a-z0-9]([-a-z0-9]*[a-z0-9])?$'
+kale_name_msg = ("must consist of lower case alphanumeric characters "
+                 "or '-', and must start and end with an alphanumeric"
+                 " character.")
+k8s_valid_name_regex = r'^[\.\-a-z0-9]+$'
```

Comment:
I think the correct regex is: `r'^[a-z][a-z0-9-.]*$'`
Plus, escaping `-` and `.` isn't required.

Reply:
@elikatsis So we should also change the `kale_name_msg` in `must consist of lower case alphanumeric characters,  '-' or '.', and must start and end with a character`
Do you also confirm that the beginning of the regex is `[a-z]` and not `[a-z0-9]`?

Reply:
Discussed live and came into this conclusion: `r'^[a-z][a-z0-9-.]*[a-z]$'`